substrate-node/pallets/pallet-dao/src/dao.rs

```diff
@@ -0,0 +1,366 @@
+    pub fn _propose(
+        who: T::AccountId,
+        threshold: u32,
+        action: Box<<T as Config>::Proposal>,
+        description: Vec<u8>,
+        link: Vec<u8>,
+        duration: Option<T::BlockNumber>,
+    ) -> DispatchResultWithPostInfo {
+        Self::is_council_member(who.clone())?;
+
+        let proposal_hash = T::Hashing::hash_of(&action);
+        ensure!(
+            !<ProposalOf<T>>::contains_key(proposal_hash),
+            Error::<T>::DuplicateProposal
+        );
+
+        let now = frame_system::Pallet::<T>::block_number();
+        let mut end = now + T::MotionDuration::get();
+        if let Some(motion_duration) = duration {
+            ensure!(
+                motion_duration < T::BlockNumber::from(constants::time::DAYS * 30),
+                Error::<T>::InvalidProposalDuration
+            );
+            end = now + motion_duration;
+        }
+
+        let index = Self::proposal_count();
+        <ProposalCount<T>>::mutate(|i| *i += 1);
+        <ProposalOf<T>>::insert(proposal_hash, *action);
+
+        let p = proposal::DaoProposal {
+            index,
+            description,
+            link,
+        };
+        <Proposals<T>>::insert(proposal_hash, p);
+
+        let votes = {
+            proposal::DaoVotes {
+                index,
+                threshold,
+                ayes: vec![],
+                nays: vec![],
+                end,
+                vetos: vec![],
+            }
+        };
+        <Voting<T>>::insert(proposal_hash, votes);
+
+        let mut active_proposal_hashes = <ProposalList<T>>::get();
+        active_proposal_hashes.push(proposal_hash);
+        <ProposalList<T>>::set(active_proposal_hashes);
+
+        Self::deposit_event(Event::Proposed {
+            account: who,
+            proposal_index: index,
+            proposal_hash,
+            threshold,
+        });
+
+        Ok(().into())
+    }
+
+    pub fn _vote(
+        who: T::AccountId,
+        farm_id: u32,
+        proposal_hash: T::Hash,
+        approve: bool,
+    ) -> DispatchResultWithPostInfo {
+        ensure!(
+            T::Tfgrid::is_farm_owner(farm_id, who.clone()),
+            Error::<T>::NotAuthorizedToVote
+        );
+
+        let stored_proposal =
+            <Proposals<T>>::get(proposal_hash).ok_or(Error::<T>::ProposalMissing)?;
+
+        let mut voting = Self::voting(proposal_hash).ok_or(Error::<T>::ProposalMissing)?;
+        ensure!(
+            voting.index == stored_proposal.index,
+            Error::<T>::WrongIndex
+        );
+
+        // Don't allow votes after the time limit reached
+        ensure!(
+            frame_system::Pallet::<T>::block_number() <= voting.end,
+            Error::<T>::TimeLimitReached
+        );
+
+        let position_yes = voting.ayes.iter().position(|a| a.farm_id == farm_id);
+        let position_no = voting.nays.iter().position(|a| a.farm_id == farm_id);
+
+        // Detects first vote of the member in the motion
```

Comment:
I think the name of the variable below properly reflects the meaning of it so you can drop this comment.

Reply:
done in 4c5945b7c0b08acb7eb1c641fd20d8f32df7559c